packages/edit-widgets/src/components/sidebar/widget-areas.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { blockDefault } from '@wordpress/icons';
+import { BlockIcon } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+
+export default function WidgetAreas( { selectedWidgetAreaId } ) {
+	const widgetAreas = useSelect(
+		( select ) => select( 'core/edit-widgets' ).getWidgetAreas(),
+		[]
+	);
+
+	const selectedWidgetArea =
+		selectedWidgetAreaId &&
+		widgetAreas?.find(
+			( widgetArea ) => widgetArea.id === selectedWidgetAreaId
+		);
+
+	let description;
+	if ( ! selectedWidgetArea ) {
+		description = __(
+			'Widget Areas are global parts in your site’s layout that can accept blocks. These vary by theme, but are typically parts like your Sidebar or Footer.'
+		);
+	} else if ( selectedWidgetAreaId === 'wp_inactive_widgets' ) {
+		description = __(
+			'Blocks in this Widget Area will not be displayed in your site.'
+		);
+	} else {
+		description = selectedWidgetArea.description;
+	}
+
+	return (
+		<div className="edit-widgets-widget-areas">
+			<div className="edit-widgets-widget-areas__top-container">
+				<BlockIcon icon={ blockDefault } />
+				<div>
+					<p>{ description }</p>
+					{ widgetAreas?.length === 0 && (
```

Comment:
What if `widgetAreas` is nullish? Do we want to display the empty message in this case?

I think that depends on whether we treat this state as a _loading_ state or an _error_ state 🤔 ?

Reply:
If we display the empty message when `widgetAreas` is nullish (`! widgetAreas || widgetAreas.length === 0`) then we'll see a flash of the empty message when the page loads, which is quite jarring. It's more common that there are widget areas so I think best to be optimistic about that.